src/tarray.jl

```diff
@@ -87,14 +87,20 @@ function Base.pop!(S::TArray)
     pop!(d)
 end
 
+function Base.convert(::Type{TArray{T, N}}, x::Array{T, N}) where {T, N}
+    return convert(TArray, x)
+end
 function Base.convert(::Type{TArray}, x::Array)
```

Comment:
Wouldn't it be more natural to define
```suggestion
function Base.convert(::Type{TArray}, x::Array)
    return convert(TArray{eltype(x),ndims(x)}, x)
end
function Base.convert(::Type{TArray{T,N}}, x::Array{T,N}) where {T,N}
```
if `convert(::Type{TArray{T,N}}, ::Array{T,N})` is needed?

Reply:
Sure.